### PR TITLE
Enable pre-commit checks on Windows

### PR DESCRIPTION
**Pull request by dthaler**
Addresses issue #2852

The path issue and solution are discussed at https://stackoverflow.com/questions/50579937/retrieve-the-path-of-a-repo-from-the-git-bash-from-windows-with-right-format/50580300#50580300

Signed-off-by: Dave Thaler <dthaler@ntdev.microsoft.com>

### CODE REVIEW

**Comment by HernanGatta** [comment]
If `shellcheck` is now a pre-requisite for Windows as well, it should be added to the [pre-requisites install script](https://github.com/dthaler/openenclave/blob/pre-commit/scripts/install-windows-prereqs.ps1), too.

File: CMakeLists.txt [hunk 1]
```diff
@@ -62,11 +62,9 @@ if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
...
 
   # Install Git pre-commit hook
-  if (NOT WIN32)
-    file(
-      COPY scripts/pre-commit scripts/commit-msg
-      DESTINATION "${PROJECT_SOURCE_DIR}/.git/hooks")
-  endif ()
+  file(
+    COPY scripts/pre-commit scripts/commit-msg
+    DESTINATION "${PROJECT_SOURCE_DIR}/.git/hooks")
```
Comment (HernanGatta) [inline CMakeLists.txt:67]: It seems odd to me to do this in CMake. If the `pre-commit` hook now runs on Windows as well, why not install the hooks by default?

Comment (jhand2) [reply CMakeLists.txt:67]: Is there such thing as hooks which are installed by default? I thought they are local to the repository so they always need to be set up either manually or by some script.

Comment (HernanGatta) [reply CMakeLists.txt:67]: Looking carefully at the Git documentation, there's a note about this which I missed on my previous read:

> It’s important to note that client-side hooks are not copied when you clone a repository. If your intent with these scripts is to enforce a policy, you’ll probably want to do that on the server side [...]

[Source](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)

So it looks like you are correct.

**Pull request merged**